feat(component): Add loading state on select and text input (#3278)
Orange-OpenSource/Orange-Boosted-Bootstrap: site/src/content/docs/forms/select-input.mdx

    </p>
  </div>`} />

### Loading

Add the `.loading-indeterminate` (for an unknown loading time) or `.loading-determinate`(for an known loading time) class on a `.select-input-field` to display a loader instead of the arrow.

> You must have the svg already in the `.select-input-container` to make it work like charm. You may also have a status span to display the loading message on assistive technologies.

Reviewer: ```suggestion
The SVG must already be in the `.select-input-container` for it to work. You should also have a status `<span>` so assistive technologies can announce the loading message.
```

When the loading starts, you will have to:
- Add the class `.loading-indeterminate` (for an unknown loading time) or `.loading-determinate` (for a known loading time) to the select to provide the appropriate styles and animation.
- For a known loading time, provide a CSS variable `--bs-loading-time` to set the loading time. For example, add `style="--bs-loading-time: 5s;"` to the select container, for a 5 seconds loading.
- Remove the `.d-none` from the status message `<span>`.
- Update regularly the status message `<span>` containing the loading message.

At the end of the loading, you should:

- Remove the class `.loading-indeterminate` or `.loading-determinate` to restore the select’s look.
- Set a final status message in `<span>`, indicating the loading has ended.
- Put the focus back on the select to ensure similar behavior across browsers and not lose the user after changes.

<Callout type="info">

>   Optionally, the select can be disabled or not during the loading time depending on the context of use.

Reviewer: Feels a little redundant to me.

```suggestion
  The select can be disabled or not during the loading time depending on the context of use.
```

The SVG must already be in the `.select-input-container` for it to work. You should also have a status `<span>` so assistive technologies can announce the loading message.

When the loading starts, you will have to:
- Add the class `.loading-indeterminate` (for an unknown loading time) or `.loading-determinate` (for a known loading time) to the select to provide the appropriate styles and animation.
- For a known loading time, provide a CSS variable `--bs-loading-time` to set the loading time. For example, add `style="--bs-loading-time: 5s;"` to the select container, for a 5 seconds loading.
- Remove the `.d-none` from the status message `<span>`.
- Update regularly the status message `<span>` containing the loading message.

At the end of the loading, you should:

- Remove the class `.loading-indeterminate` or `.loading-determinate` to restore the select’s look.
- Set a final status message in `<span>`, indicating the loading has ended.
- Put the focus back on the select to ensure similar behavior across browsers and not lose the user after changes.

<Callout type="info">

>   The select can be disabled or not during the loading time depending on the context of use.

Reviewer 2: We say at the beginning of the states section that states are exclusive and here we say they can be mixed :)
